druid-shell/src/platform/x11/window.rs

```diff
@@ -763,15 +816,79 @@ impl Window {
     }
 
     /// Set whether the window should be resizable
-    fn resizable(&self, _resizable: bool) {
-        warn!("Window::resizeable is currently unimplemented for X11 platforms.");
+    fn resizable(&self, resizable: bool) {
+        let conn = self.app.connection().as_ref();
+        log_x11!(size_hints(resizable, self.size().size_px(), self.min_size)
+            .set_normal_hints(conn, self.id)
+            .context("set normal hints"));
     }
 
     /// Set whether the window should show titlebar
     fn show_titlebar(&self, _show_titlebar: bool) {
         warn!("Window::show_titlebar is currently unimplemented for X11 platforms.");
     }
 
+    fn get_position(&self) -> Point {
+        fn _get_position(window: &Window) -> Result<Point, Error> {
+            let conn = window.app.connection();
+            let scale = window.scale.get();
+            let geom = conn.get_geometry(window.id)?.reply()?;
+            let cord = conn
+                .translate_coordinates(window.id, geom.root, geom.x, geom.y)?
+                .reply()?;
+            Ok(Point::new(cord.dst_x as _, cord.dst_y as _).to_dp(scale))
+        }
+        let pos = _get_position(self);
+        log_x11!(&pos);
+        pos.unwrap_or_default()
+    }
+
+    fn set_position(&self, pos: Point) {
+        let conn = self.app.connection();
+        let scale = self.scale.get();
+        let pos = pos.to_px(scale).expand();
+        log_x11!(conn.configure_window(
+            self.id,
+            &ConfigureWindowAux::new().x(pos.x as i32).y(pos.y as i32),
+        ));
+    }
+
+    fn set_size(&self, size: Size) {
+        let conn = self.app.connection();
+        let scale = self.scale.get();
+        let size = size.to_px(scale).expand();
+        log_x11!(conn.configure_window(
+            self.id,
+            &ConfigureWindowAux::new()
+                .width(size.width as u32)
+                .height(size.height as u32),
+        ));
+    }
+
+    fn set_level(&self, level: WindowLevel) {
+        let window_type = match level {
+            WindowLevel::AppWindow => self.atoms._NET_WM_WINDOW_TYPE_NORMAL,
+            WindowLevel::Tooltip => self.atoms._NET_WM_WINDOW_TYPE_TOOLTIP,
+            WindowLevel::Modal => self.atoms._NET_WM_WINDOW_TYPE_DAILOG,
+            WindowLevel::DropDown => self.atoms._NET_WM_WINDOW_TYPE_DROPDOWN_MENU,
+        };
+
+        let conn = self.app.connection();
+        log_x11!(conn.change_property32(
+            xproto::PropMode::REPLACE,
+            self.id,
+            self.atoms._NET_WM_WINDOW_TYPE,
+            AtomEnum::ATOM,
+            &[window_type],
+        ));
+        if let WindowLevel::DropDown = level {
```

Comment:
We probably want this for tooltip and modal also? At least, that's what I did in the gtk backend.

Reply:
Done